packages/commons-server/src/libs/server/server.ts

```diff
@@ -585,6 +591,100 @@ export class MockoonServer extends (EventEmitter as new () => TypedEmitter<Serve
     };
   }
 
+  private makeCallbacks(
+    route: Route,
+    routeResponse: RouteResponse,
+    request: Request,
+    response: Response
+  ) {
+    if (routeResponse.callbacks && routeResponse.callbacks.length > 0) {
+      try {
+        for (const invocation of routeResponse.callbacks) {
+          const cb = this.environment.callbacks.find(
+            (ref) => ref.uuid === invocation.uuid
+          );
+
+          if (!cb) {
+            continue;
+          }
+
+          const url = TemplateParser(
+            false,
+            cb.uri,
+            this.environment,
+            this.processedDatabuckets,
+            request,
+            response
+          );
+
+          let content = cb.body;
+          if (cb.bodyType === BodyTypes.INLINE) {
+            content = TemplateParser(
+              false,
+              content || '',
+              this.environment,
+              this.processedDatabuckets,
+              request,
+              response
+            );
+          } else if (cb.bodyType === BodyTypes.DATABUCKET && cb.databucketID) {
+            const servedDatabucket = this.processedDatabuckets.find(
+              (processedDatabucket) =>
+                processedDatabucket.id === cb.databucketID
+            );
+
+            if (servedDatabucket) {
+              content = servedDatabucket.value;
+
+              // if returned content is an array or object we need to stringify it for some values (array, object, booleans and numbers (bool and nb because expressjs cannot serve this as is))
+              if (
+                Array.isArray(content) ||
+                typeof content === 'object' ||
+                typeof content === 'boolean' ||
+                typeof content === 'number'
+              ) {
+                content = JSON.stringify(content);
+              } else {
+                content = content;
+              }
+            }
+          } else if (cb.bodyType === BodyTypes.FILE && cb.filePath) {
+            this.sendFileWithCallback(
+              route,
+              routeResponse,
+              cb,
+              invocation,
+              request,
+              response
+            );
+
+            continue;
+          }
+
+          const sendingHeaders = {
+            headers: {}
+          };
+          this.setHeaders(cb.headers || [], sendingHeaders, request);
+
+          setTimeout(() => {
+            fetch(url, {
+              method: cb.method,
+              headers: sendingHeaders.headers,
+              body: content
+            });
```

Comment:
You should deal with fetch Promise with a then and a catch. Otherwise we won't be able to surface errors and it will throw UnhandledPromiseRejectionWarning. 

Something like:

```
fetch(url, {
              method: cb.method,
              headers: sendingHeaders.headers,
              body: content
            })
              .then(() => {})
              .catch((e) => {
                this.emit('error', ServerErrorCodes.CALLBACK_ERROR, e);
              });
```
This helped me surface this error, which is probably why most methods are not working: 
`Request with GET/HEAD method cannot have body.!` for GET or HEAD
Or also: 
`fetch failed!` for a PATCH

Another thing, in the then, a new event could be emitted, like the `transaction-complete`, maybe `callback-executed`, so it would get logged too.

Reply:
Great to know about this!

Reply:
I created a new event called `callback-invoked` so that callback invocations will be logged.